Microsoft.Toolkit.Uwp/Helpers/BackgroundTaskHelper.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            return BackgroundTaskRegistration.AllTasks.Any(t => t.Value.Name == backgroundTaskName);
+        }
+
+        /// <summary>
+        /// Check if a background task is registered.
+        /// </summary>
+        /// <param name="backgroundTaskType">The type of the background task. This class has to implement IBackgroundTask</param>
+        /// <returns>True/False indicating if a background task was registered or not</returns>
+        public static bool IsBackgroundTaskRegistered(Type backgroundTaskType)
+        {
+            return IsBackgroundTaskRegistered(backgroundTaskType.Name);
+        }
+
+        /// <summary>
+        /// Register a background task with conditions.
+        /// If the task is already registered and has the same trigger, returns the existing registration if it has the same trigger.
+        /// If the task is already registered but has different trigger, return null by default.
+        /// Or set <paramref name="forceRegister"/> to true to un-register the old one and then re-register.
+        /// </summary>
+        /// <param name="backgroundTaskName">Name of the background task class</param>
+        /// <param name="backgroundTaskEntryPoint">Entry point of the background task. This parameter will not be used if <paramref name="isSingleProcess"/> is true</param>
+        /// <param name="trigger">Trigger that indicate when the background task should be invoked</param>
+        /// <param name="forceRegister">Indicate if the background task will be force installed in the case of being already registered</param>
+        /// <param name="isSingleProcess">Indicate if we are registering under Single Process Model. <paramref name="backgroundTaskEntryPoint"/> won't be used if this is true. Default is false.
+        /// WARNING: Single Process Model only works with Windows 10 Anniversary Update (14393) and later.</param>
+        /// <param name="enforceConditions">Indicate if the background task should quit if condition is no longer valid</param>
+        /// <param name="conditions">Optional conditions for the background task to run with</param>
+        /// <returns>Background Task that was registered with the system</returns>
+        public static BackgroundTaskRegistration Register(string backgroundTaskName, string backgroundTaskEntryPoint, IBackgroundTrigger trigger, bool forceRegister = false, bool isSingleProcess = false, bool enforceConditions = true, params IBackgroundCondition[] conditions)
+        {
+            // Check if the task is already registered.
+            if (IsBackgroundTaskRegistered(backgroundTaskName) == true)
+            {
+                BackgroundTaskRegistration previouslyRegistered = GetBackgroundTask(backgroundTaskName) as BackgroundTaskRegistration;
+
+                if (forceRegister == false)
+                {
+                    Unregister(previouslyRegistered);
+                }
+                else if (previouslyRegistered.Trigger == trigger)
```

Comment:
Will this comparison work? Isn't this going to end up being a reference comparison, which will be false since I construct a new TimeTrigger each time, even though the values are the same? Or am I missing something here like I was with the FirstOrDefault thing :)